File: 0206-reverse-linked-list/0206-reverse-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* reverseList(struct ListNode* head) {
    if(head==NULL)
        return NULL;
    struct ListNode *prev=NULL;
    struct ListNode *temp=head;
    struct ListNode *next=NULL;
    while(temp!=NULL)
    {
        next=temp->next;
        temp->next=prev;

        prev=temp;
        temp=next;
        
    } 
    head=prev;
    return head;
}

File: 0206-reverse-linked-list/0206-reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        if(head==null) return null;
        ListNode prev = null;
        ListNode temp = head;
        ListNode next_node = null;
        while(temp!=null){
            next_node = temp.next;
            temp.next=prev;
            prev=temp;
            temp=next_node;
            
        }
        head=prev;
        return head;
    }
}